wgpu-hal/src/vulkan/instance.rs

```diff
@@ -172,29 +175,40 @@ impl super::Instance {
             not(target_os = "android"),
             not(target_os = "macos")
         )) {
+            // VK_KHR_xlib_surface
             extensions.push(khr::XlibSurface::name());
+            // VK_KHR_xcb_surface
             extensions.push(khr::XcbSurface::name());
+            // VK_KHR_wayland_surface
             extensions.push(khr::WaylandSurface::name());
         }
         if cfg!(target_os = "android") {
+            // VK_KHR_android_surface
             extensions.push(khr::AndroidSurface::name());
         }
         if cfg!(target_os = "windows") {
+            // VK_KHR_win32_surface
             extensions.push(khr::Win32Surface::name());
         }
         if cfg!(target_os = "macos") {
+            // VK_EXT_metal_surface
             extensions.push(ext::MetalSurface::name());
         }
 
         if flags.contains(crate::InstanceFlags::DEBUG) {
+            // VK_EXT_debug_utils
             extensions.push(ext::DebugUtils::name());
         }
 
-        extensions.push(vk::KhrGetPhysicalDeviceProperties2Fn::name());
-
+        // VK_EXT_swapchain_colorspace
         // Provid wide color gamut
         extensions.push(vk::ExtSwapchainColorspaceFn::name());
 
+        // VK_KHR_get_physical_device_properties2
+        if driver_api_version < vk::API_VERSION_1_1 {
```

Comment:
Because we only enable this on vulkan 1 - we end up failing to load the function pointers on vk1.1+ (at least on my machine). If we unconditionally add the extension, everything works fine.

Reply:
So, this is not working properly?

https://github.com/gfx-rs/wgpu/blob/61d375ed0db2e6df18963537b59e478dcba0e2a4/wgpu-hal/src/vulkan/instance.rs#L279-L280

Reply:
I'm not sure what you mean - the link is to the same code I pointed at.

Reply:
Sorry read that first thing in the morning on my phone where they don't give me highlighted line numbers 😆 

It does work - the problem is that https://github.com/gfx-rs/wgpu/blob/61d375ed0db2e6df18963537b59e478dcba0e2a4/wgpu-hal/src/vulkan/instance.rs#L283-L286 only gets the proper pointer if the instance extension is enabled, which it isn't on VK1.1+

Reply:
Ah, so even though the [`VK_KHR_get_physical_device_properties2`](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_get_physical_device_properties2.html) extension became part of core Vulkan 1.1 with no changes, the function pointers are still different between the extension and core.

I now see why we we still require `VK_KHR_draw_indirect_count` even for vk 1.2+; so that we can avoid having to load different function pointers.

Are there any downsides to this approach?

Reply:
> the function pointers are still different between the extension and core.

The function pointers you get are likely the same, the loader is just being correct and only giving you the pointer by the KHR name if you've enabled the extension. 

AFAIK there are no downsides for requiring the extension - I believe extensions are always required to be supported if the functionality is supported through core. Even if it's not required, I don't think I've ever seen it not be the case.